### PR TITLE
Document setting box.cfg parameters via environment variables

### PR DESCRIPTION
**Pull request by veod32**

### CODE REVIEW

**Comment by Totktonada** [comment]
I suggest to add environment variable names into box.cfg parameters description explicitly for user's convenience. Looks at the etcd documentation for inspiration: https://etcd.io/docs/v3.4/op-guide/configuration/.

**Comment by veod32** [comment]
> I suggest to add environment variable names into box.cfg parameters description explicitly for user's convenience. Looks at the etcd documentation for inspiration: https://etcd.io/docs/v3.4/op-guide/configuration/.

Good suggestion. Currently, there're around 75 parameters in the config reference. If we're going to do this update, I think it's better to do it as a separate task.

**Comment by Totktonada** [comment]
Okay. Let's split it as it is convenient for you.

**Comment by veod32** [comment]
> Okay. Let's split it as it is convenient for you.

#2722

File: doc/reference/configuration/index.rst [hunk 1]
```diff
@@ -296,66 +300,106 @@ dynamic, that is, they can be changed at runtime by calling ``box.cfg{}``
...
+
+..  _box-cfg-params-prior:
+
+Methods of setting and priorities
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tarantool configuration parameters can be specified in different ways.
+The priority of parameter sources is the following, from higher to lower:
+
+*   ``box.cfg{}`` call
```
Comment (Totktonada) [inline doc/reference/configuration/index.rst:313]: `box.cfg{}` call options, maybe? To make it clear and strictly separated from defaults.

File: doc/reference/configuration/index.rst [hunk 2]
```diff
@@ -296,66 +300,106 @@ dynamic, that is, they can be changed at runtime by calling ``box.cfg{}``
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Starting from version :doc:`2.8.1 </release/2.8.1>`, you can specify configuration parameters via special environment variables.
+The name of a variable should have the following pattern: ``TT_<NAME>``,
+where ``<NAME>`` is the uppercase name of the corresponding :ref:`box.cfg parameter <box-cfg-params-ref>`.
+
+For example:
+
+* ``TT_LISTEN``---corresponds to the ``box.cfg.listen`` option.
+* ``TT_REPLICATION``---corresponds to the ``box.cfg.replication`` option.
```
Comment (Totktonada) [inline doc/reference/configuration/index.rst:330]: It worth to add one example of an option, whose name is constructed from several words.

**Review by Totktonada — APPROVED** [review]
LGTM. Thanks!

File: doc/reference/configuration/index.rst (hunk 1 above)
Comment (veod32) [reply doc/reference/configuration/index.rst:313]: Corrected

File: doc/reference/configuration/index.rst (hunk 2 above)
Comment (veod32) [reply doc/reference/configuration/index.rst]: Added ``memtx_dir`` as an example of such options.

**Pull request merged**